aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferStatusUpdater.java

```diff
@@ -145,6 +151,15 @@ synchronized void removeTransfer(int id) {
      * @param id id of the transfer to remove
      */
     synchronized void removeTransferRecordFromDB(final int id) {
+        // Remove temporary file
+        TransferRecord transferRecord = dbUtil.getTransferById(id);
+        if (transferRecord != null) {
+            String path = transferRecord.file;
+            String fileName = new File(path).getName();
+            if (fileName.startsWith(TEMP_FILE_PREFIX)) {
+                new File(path).delete();
+            }
+        }
```

Comment:
This part worries me, because if a customer names their file `aws-s3-my-file`, then `TransferUtility` will delete it after upload even if it isn't temporary. Accidentally deleting customer data is probably a bigger issue than leaving a temp file sitting in their memory in my opinion. Is there a better way to mark a file as temporary?

Reply:
I think maybe we could do the check with both the prefix and the suffix which is ".tmp". It's less possible for customers to upload a file with this kind of suffix. Or, make the prefix something like "aws-s3-generated-for-inputstream"?

Reply:
Appending a constant UUID to the prefix would probably be a reasonable solution (e.g. `aws-s3-d861b25a-1edf-11eb-adc1-0242ac120002`)

I think it's actually quite reasonable that a customer could upload a file with the suffix `.tmp`, because that's the default suffix if none is provided when calling `File.createTempFile`, so I wouldn't rely on that.

Reply:
When do checking before deleting a temp file, we don't know the UUID exactly. But if a random UUID indeed follows the format like `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`, we can adopt this method by checking whether the temp file fits into the format.

Reply:
To clarify, I'm suggesting that you simply replace:
```
static final String TEMP_FILE_PREFIX = "aws-s3";
```
with 

```
static final String TEMP_FILE_PREFIX = "aws-s3-d861b25a-1edf-11eb-adc1-0242ac120002";
```

The chance that a customer would also prefix a file with `aws-s3` is very low -- but the chance they would prefix a File with `aws-s3-d861b25a-1edf-11eb-adc1-0242ac120002` is virtually none.

Reply:
Got it. I thought each time we would generate a new UUID. Thanks! This totally make sense.